upf/upf.c

```diff
@@ -160,6 +163,15 @@ vnet_upf_nat_pool_add_del (u8 * nwi_name, ip4_address_t start_addr,
       if (p)
 	return VNET_API_ERROR_VALUE_EXIST;
 
+      min_port = clib_max (UPF_NAT_MIN_PORT, min_port);
```

Comment:
I think you should unconditionally overwrite the min/max port. Instead check and report an error if they exceed the limits.

Reply:
Fixed